### PR TITLE
feat: update paloma app to SDK v0.50

### PR DESCRIPTION
**Pull request by deepan95dev**
# Background

Upgraded the files to 0.50 other than x

# Testing completed

- [x] No Import errors
- [x] No Usage errors

### CODE REVIEW

File: app/export.go [hunk 1]
```diff
@@ -16,7 +16,7 @@ import (
 // file.
 func (app *App) ExportAppStateAndValidators(forZeroHeight bool, jailAllowedAddrs, modulesToExport []string) (servertypes.ExportedApp, error) {
 	// as if they could withdraw from the start of the next block
-	ctx := app.NewContext(true, tmproto.Header{Height: app.LastBlockHeight()})
+	ctx := app.NewContext(true /*tmproto.Header{Height: app.LastBlockHeight()}*/)
```
Comment (aleem1314) [inline app/export.go:19]: Why second argument commented?

File: cmd/palomad/root.go [hunk 2]
```diff
@@ -185,28 +192,83 @@ func rootPreRunE(cmd *cobra.Command, args []string) error {
...
+}
+
+// newApp creates the application
+func newApp(
+	logger log.Logger,
+	db dbm.DB,
+	traceStore io.Writer,
+	appOpts servertypes.AppOptions,
+) servertypes.Application {
+	//baseappOptions := server.DefaultBaseappOptions(appOpts)
```
Comment (aleem1314) [inline cmd/palomad/root.go:243]: ditto here

File: app/app.go [hunk 3]
```diff
@@ -8,31 +8,45 @@ import (
 	"path/filepath"
 	"strings"
 
-	"github.com/CosmWasm/wasmd/x/wasm"
-	wasmkeeper "github.com/CosmWasm/wasmd/x/wasm/keeper"
-	dbm "github.com/cometbft/cometbft-db"
+	// storetypes "cosmossdk.io/store/types"
```
Comment (aleem1314) [inline app/app.go:11]:
```suggestion
```

File: app/app.go [hunk 4]
```diff
@@ -41,9 +55,13 @@ import (
 	"github.com/cosmos/cosmos-sdk/x/bank"
 	bankkeeper "github.com/cosmos/cosmos-sdk/x/bank/keeper"
 	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
-	"github.com/cosmos/cosmos-sdk/x/capability"
-	capabilitykeeper "github.com/cosmos/cosmos-sdk/x/capability/keeper"
-	capabilitytypes "github.com/cosmos/cosmos-sdk/x/capability/types"
+
+	"github.com/cosmos/ibc-go/modules/capability"
+	// ica "github.com/cosmos/ibc-go/v3/modules/apps/27-interchain-accounts"
```
Comment (aleem1314) [inline app/app.go:60]: ditto here

File: cmd/palomad/root.go [hunk 5]
```diff
@@ -93,7 +100,7 @@ func NewRootCmd() *cobra.Command {
 		flags.FlagKeyringBackend: keyring.BackendOS,
 	}); err != nil {
 		// seems to be non-fatal issue
-		log.Printf("error overriding flag defaults: %s", err.Error())
+		fmt.Println("error overriding flag defaults: %s", err.Error())
```
Comment (byte-bandit) [inline cmd/palomad/root.go:103]: Why not use the log package?

File: app/export.go (hunk 1 above)
Comment (vishal-kanna) [reply app/export.go:19]: NewContext takes only one argument which is of type bool.

File: cmd/palomad/root.go (hunk 5 above)
Comment (deepan95dev) [reply cmd/palomad/root.go:103]: We imported `"cosmossdk.io/log"`  which does not have Printf function so its updated to fmt.

Comment (deepan95dev) [reply cmd/palomad/root.go]: pushed the changes with "log" module

**Comment by aleem1314** [comment]
Looks good to me. Let's update each Paloma module in its own pull request and then integrate them with the Paloma app.

File: cmd/palomad/root.go (hunk 5 above)
Comment (byte-bandit) [reply cmd/palomad/root.go]: We don't encourage using the `fmt()` package for logging. You can use `log.Info()` instead.

**Pull request merged**